### DISCUSSION TITLE
footnotes usage with `gatsby-transformer-remark`

### DISCUSSION THREAD

**Discussion by PsyGik** [remarkjs/remark#737] [Q&A] [answered]
Hello 👋🏽,

Cross-posting this from [`gatsby-transformer-remark #30964`
](https://github.com/gatsbyjs/gatsby/issues/30964), as [suggested by a contributor](https://github.com/gatsbyjs/gatsby/issues/30964#issuecomment-841835090)

I am not sure if this change broke the implementation, but this no longer works:

```
Some Markdown Content[^1]
Another Content
Some More Content[^2]


[^1]: First Footnote
[^2]: Second Footnote
[^3]: Third Footnote // This doesn't get rendered.
```
This outputs as:

```
<p>
    Some Markdown Content<sup id="fnref-1"><a href="#fn-1" class="footnote-ref">1</a></sup>
    Another Content
    Some More Content<sup id="fnref-2"><a href="#fn-2" class="footnote-ref">2</a></sup>
</p>
<div class="footnotes">
    <hr>
    <ol>
    <li id="fn-1">First Footnote<a href="#fnref-1" class="footnote-backref">↩</a></li>
    <li id="fn-2">Second Footnote<a href="#fnref-2" class="footnote-backref">↩</a></li>
    </ol>
</div>
```
The `[^3]` isn't referenced anywhere in the content, so it doesn't get rendered.

As soon as I add it to the markdown, it get's rendered.
```

Some Markdown Content[^1]
Another Content[^3]
Some More Content[^2]


[^1]: First Footnote
[^2]: Second Footnote
[^3]: Third Footnote // This doesn't get rendered.
```

Is this the intended behaviour? Because on "gatsby": "^3.2.1", footnotes used to render without being referenced anywhere else in the markdown

**Comment by wooorm** [comment, marked as answer]
Yes, this behavior is intended. Here’s the change plus links to discussions: https://github.com/syntax-tree/mdast-util-to-hast/commit/fd38c45421bbec497f56e5c624eb8652d3a3bba4
